src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js

```diff
@@ -160,6 +190,61 @@ class ActivityFeed extends React.Component<Props, State> {
         return feedItems.length === 0 || (feedItems.length === 1 && feedItems[0].type === ItemTypes.fileVersion);
     };
 
+    getActiveEntry = (callback: (activeEntry: FeedItem | null, repliesNeedToBeLoaded?: boolean) => void) => {
+        const { activeFeedEntryId, activeFeedEntryType, feedItems, getComment, hasReplies } = this.props;
+
+        if (!activeFeedEntryId || !Array.isArray(feedItems)) {
+            return callback(null);
+        }
+
+        const firstLevelItem = feedItems.find(
+            ({ id, type }) => id === activeFeedEntryId && type === activeFeedEntryType,
+        );
+        // If the active entry is a first level Feed item, return it
+        if (firstLevelItem) {
+            return callback(firstLevelItem);
```

Comment:
We generally want to avoid multiple returns, as it makes a function very difficult to debug and reason about. This function is also _very_ complicated and probably needs to be decomposed.